tests/config/test_omegaconf_config.py

```diff
@@ -649,3 +649,21 @@ def test_variable_interpolation_in_catalog_with_separate_templates_file(
         conf = OmegaConfigLoader(str(tmp_path))
         conf.default_run_env = ""
         assert conf["catalog"]["companies"]["type"] == "pandas.CSVDataSet"
+
+    def test_custom_resolvers(self, tmp_path):
+        base_params = tmp_path / _BASE_ENV / "parameters.yml"
+        param_config = {
+            "model_options": {
+                "test_size": "${add: 3, 4}",
+                "random_state": "${plus_2: 1}",
+            }
+        }
+        _write_yaml(base_params, param_config)
+        custom_resolvers = {
+            "add": lambda *x: sum(x),
+            "plus_2": lambda x: x + 2,
+        }
+        conf = OmegaConfigLoader(str(tmp_path), custom_resolvers=custom_resolvers)
```

Comment:
is `str(tmp_path)` strictly needed?

Reply:
I think so, `conf_source` is required argument

Reply:
Sorry, I mean if `tmp_path` alone works? I check the definition of `AbstractConfigLoader` have a signature of `conf_source: str`, but I expect it should be `conf_source : str | Path`.
![image](https://github.com/kedro-org/kedro/assets/18221871/140dbd86-cca9-4a5f-8dd9-3520b868c619)

A quick experiment suggests that it should work

![image](https://github.com/kedro-org/kedro/assets/18221871/2a53b31e-07a2-4b9e-a743-aead27de6f6f)

cc @merelcht